main.go

```diff
@@ -10,625 +10,169 @@
-							}
-							filePathMap[path] = fileMD5
-						} else if indicator == MODTIME {
-							filePathMap[path] = fmt.Sprintf("%d", info.ModTime().Unix())
-						}
-					case SKIP:
-						if debug {
-							log.Printf("  Ignore changes: %s", path)
-						}
-						filePathMap[path] = "-"
-					case INDICATOR:
-						filePathMap[path] = indicatorFileMD5
-					}
-				}
-				if archiveFiles {
-					file, err := os.Open(path)
-					if err != nil {
-						return err
-					}
-
-					defer func() {
-						if err := file.Close(); err != nil {
-							log.Errorf("Failed to close file (%s), error: %+v", path, err)
-						}
-					}()
-
-					_, err = io.CopyN(writer, file, info.Size())
-					if err != nil && err != io.EOF {
-						return err
-					}
-				}
-			}
-			return nil
-		}); err != nil {
-			return nil, err
-		}
-	}
-	return filePathMap, nil
+func logErrorfAndExit(format string, args ...interface{}) {
+	log.Errorf(format, args...)
+	os.Exit(1)
 }
 
-// GetStoreMode ...
-func GetStoreMode(indicatorHashMap map[string]string, ignoreList []string, path string) (StoreMode, string) {
-	for key, value := range indicatorHashMap {
-		if strings.HasPrefix(path, key) {
-			return INDICATOR, value
-		}
-	}
-
-	for _, ignoreFilter := range ignoreList {
-		ignoreFromArchive := strings.HasPrefix(ignoreFilter, "!")
-		cleanedIgnoreFilter := strings.TrimSpace(strings.TrimPrefix(ignoreFilter, "!"))
-
-		if strings.Contains(cleanedIgnoreFilter, "*") {
-			if glob.Glob(cleanedIgnoreFilter, path) {
-				if ignoreFromArchive {
-					return REMOVE, ""
-				}
-				return SKIP, ""
-			}
-		} else {
-			if strings.HasPrefix(path, cleanedIgnoreFilter) {
-				if ignoreFromArchive {
-					return REMOVE, ""
-				}
-				return SKIP, ""
-			}
-		}
-	}
-	return STORE, ""
-}
+func main() {
+	stepStartedAt := time.Now()
 
-// LoadPreviousFilePathMap ...
-func LoadPreviousFilePathMap() (map[string]string, bool, error) {
-	exists, err := pathutil.IsPathExists(cacheInfoFilePath)
+	configs, err := ParseConfig()
 	if err != nil {
-		return nil, false, err
-	}
-	if !exists {
-		return nil, false, nil
+		log.Errorf(err.Error())
+		os.Exit(1)
 	}
 
-	fileBytes, err := fileutil.ReadBytesFromFile(cacheInfoFilePath)
-	if err != nil {
-		return nil, false, err
-	}
+	configs.Print()
+	fmt.Println()
 
-	var previousFilePathMap map[string]string
-	err = json.Unmarshal(fileBytes, &previousFilePathMap)
-	if err != nil {
-		return nil, false, err
-	}
+	// Cleaning paths
+	startTime := time.Now()
 
-	return previousFilePathMap, true, nil
-}
+	log.Infof("Cleaning paths")
 
-// CompareFilePathMaps ...
-func CompareFilePathMaps(previousFilePathMap, currentFilePathsMap map[string]string, debugMode bool) bool {
-	logDebugPaths := func(paths []string) {
-		if debugMode {
-			for _, pth := range paths {
-				log.Debugf("- %s", pth)
-			}
-		}
+	indicatorByPth := parseIncludeList(strings.Split(configs.Paths, "\n"))
+	if len(indicatorByPth) == 0 {
+		log.Warnf("No path to cache, skip caching...")
```

Comment:
Maybe you could a similar method like `logErrorfAndExit()` this.
Or modify the `logErrorfAndExit()` method to 
- log a warning message and exit with exit code 0 or
-  log an error message and exit with exit code 1

Reply:
i will leave it as it is for now, we use `os.Exit(0)` in 3 places.
For 2 we could create a share function but for the 3. we use different logging.